Source/BugsnagConfiguration.m

```diff
@@ -504,4 +521,19 @@ - (void)addPlugin:(id<BugsnagPlugin> _Nonnull)plugin {
     [_plugins addObject:plugin];
 }
 
+- (void)notifyObserver:(BugsnagStateEvent *)event {
+    if (self.block != nil) {
+        self.block(event);
+    }
+}
+
+- (void)registerStateObserverWithBlock:(BugsnagObserverBlock _Nonnull)block {
+    self.block = block;
+}
+
+- (void)metadataChanged:(BugsnagMetadata *_Nonnull)metadata {
```

Comment:
Should/could `BugsnagMetadata` do this directly rather than delegating to `Configuration` to notify observers?